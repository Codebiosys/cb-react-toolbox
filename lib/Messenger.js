'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _reactRedux = require('react-redux');

var _reactNotificationSystemRedux = require('react-notification-system-redux');

var _reactNotificationSystemRedux2 = _interopRequireDefault(_reactNotificationSystemRedux);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var mapStateToProps = function mapStateToProps(state) {
  return {
    notifications: state.notifications
  };
};

/**
 * @typedef {Class} Messenger
 *
 * @desc Notification management via redux. Include the notification in the outer
 * scope of your application, and then dispatch the appropriate actions.
 *
 * @see https://github.com/gor181/react-notification-system-redux
 */
exports.default = (0, _reactRedux.connect)(mapStateToProps)(_reactNotificationSystemRedux2.default);