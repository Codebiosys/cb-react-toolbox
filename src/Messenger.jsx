import { connect } from 'react-redux';
import Notifications from 'react-notification-system-redux';

const mapStateToProps = state => ({
  notifications: state.notifications,
});

/**
 * @typedef {Class} Messenger
 *
 * @desc Notification management via redux. Include the notification in the outer
 * scope of your application, and then dispatch the appropriate actions.
 *
 * @see https://github.com/gor181/react-notification-system-redux
 */
export default connect(mapStateToProps)(Notifications);
