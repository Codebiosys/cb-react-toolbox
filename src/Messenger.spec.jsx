import React from 'react';
import { shallow } from 'enzyme';
import { createMockStore } from 'redux-test-utils';
import PropTypes from 'prop-types';

import Notifications from 'react-notification-system-redux';

import Messenger from './Messenger';

describe('The Messenger notifier', () => {
  let wrapper;
  let mounted;
  const defaultStore = { notifications: [{ title: 'A Message' }] };
  let store;
  beforeEach(() => {
    store = createMockStore(defaultStore);
    mounted = shallow(
      <Messenger />,
      { context: { store },
        childContextTypes: { store: PropTypes.object },
      },
    );
    wrapper = mounted.find(Notifications);
  });

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('has current state notifications', () => {
    expect(wrapper.props()).toHaveProperty('notifications');
    expect(wrapper.props().notifications).toEqual(defaultStore.notifications);
  });
});
