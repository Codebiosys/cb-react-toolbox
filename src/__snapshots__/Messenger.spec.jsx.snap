// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Messenger notifier renders 1`] = `
<Notifications
  dispatch={[Function]}
  notifications={
    Array [
      Object {
        "title": "A Message",
      },
    ]
  }
/>
`;
